main.go

```diff
@@ -0,0 +1,219 @@
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"sort"
+)
+
+func UI() (action uint) {
+	fmt.Println(`To choose your action enter number:
+	1) List all entries.
+	2) Add new entry.
+	3) Remove an entry by ID.
+	4) Exit
+	5) Leave feedback`)
+	fmt.Scan(&action)
+	if action > 5 || action == 0 {
+		fmt.Println("No such action, please restart program.")
+	}
+	return action
+}
+
+type Entry struct {
+	ID          uint
+	FirstName   string
+	LastName    string
+	PhoneNumber string
+}
+
+func Sort(entries []Entry) {
+	sort.Slice(entries, func(i, j int) bool { return entries[i].LastName < entries[j].LastName })
+}
+
+func Header() {
+	fmt.Printf("%10s %20s %20s %20s\n", "ID", "Last Name", "First Name", "Phone Number")
+	fmt.Println("-------------------------------------------------------------------------")
+}
+func ReadPhonebook(phonebookFile string) (entries []Entry, err error) {
+	f, err := os.Open(phonebookFile)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+	r := bufio.NewReader(f)
+	if err = json.NewDecoder(r).Decode(&entries); err != nil {
+		return nil, err
+	}
+	return entries, nil
+}
+func ReadFeedback() (feedbacks []Feedbackstruct, err error) {
+	f, err := os.Open("feedback.json")
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+	r := bufio.NewReader(f)
+	if err = json.NewDecoder(r).Decode(&feedbacks); err != nil {
+		return nil, err
+	}
+	return feedbacks, nil
+}
+func List(phonebookFile string) {
+	var entries []Entry
+	entries, _ = ReadPhonebook(phonebookFile)
+	Sort(entries)
+	Header()
+	for i, j := range entries {
+		fmt.Printf("%10d %20s %20s %20s\n", j.ID, j.LastName, j.FirstName, j.PhoneNumber)
+		if (i+1)%20 == 0 && i < len(entries) {
+			fmt.Println(" ")
+			fmt.Println("Press <ENTER> to continue...")
+			fmt.Println(" ")
+			fmt.Scanln()
+			Header()
+		}
+	}
+}
+
+func GiveID(entries []Entry) (i uint) {
+	/*for i, j := range entries { // i dont know how to combine it, also in this for i always is 1
+		j.ID = uint(i) + 1
+	}
+	return uint(i) + 1*/
+	i = uint(len(entries)) + 1
```

Comment:
This still isn't fully correct. You have to find max ID and your next ID is max + 1.